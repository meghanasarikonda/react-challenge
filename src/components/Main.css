.box-enter {
  opacity: 0;
  animation: bounceup 200ms ease-in;
}
.box-enter-active {
  opacity: 1;
  animation: bounceup 250ms ease-in;
}
.box-exit {
  opacity: 1;
  animation: fadeout 200ms ease-out;
}
.box-exit-active {
  opacity: 0;
  animation: fadeout 200ms ease-out;
}


@keyframes bounceup {
  0% {
    transform: scale(0.1);
    opacity: 0;
  }
  60% {
    transform: scale(1.1);
    opacity: 1;
  }
  100% {
    transform: scale(1);
    opacity: 1;
  }
}

@keyframes fadeout {
  0% {
    transform: scale(1.0);
    opacity: 1;
  }

  60% {
    opacity: 0;
    transform: scale(0.4);
  }

  100% {
    opacity: 0;
    transform: scale(0);
  }
}