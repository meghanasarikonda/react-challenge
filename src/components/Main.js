import React, { Component } from 'react';
import ImageList from '../Imagefiles.js';
import SingleAvatar from './SingleAvatar';
import AvatarList from './AvatarList';
import { CSSTransition } from 'react-transition-group';


class Main extends Component {

  constructor(props) {
    super(props);
    this.state = {
      image: ImageList[0],
      ImageList: ImageList,
      openAvatarList: false,
      isLoading: false
    }

    this.imageUpdater = (id) => {
      //console.log(e, 'e from line 16')
      this.setState({
        image: ImageList[id - 1]
      });
    }

    this.openPopover = () => {
      this.setState({
        openAvatarList: true
      })
    }

    this.loadingChecker = (bool) => {
      this.setState({
        isLoading: bool
      })

      this.closePopover = () => {
        this.setState({
          openAvatarList: false
        })
      }

    }

    this.singleavatar = () => {
      console.log('line 44 singleavatar')
      setTimeout(() => {
        if (!this.state.isLoading) {
          this.setState({
            openAvatarList: false
          })
        }
      }, 200)
    }

  }


  render() {

    return (
      <div>

        <SingleAvatar image={this.state.image} openPopover={this.openPopover} />
        <CSSTransition
          in={this.state.openAvatarList}
          classNames="message"
          timeout={300}
          unmountOnExit
        >

        <AvatarList imageObj={this.state.ImageList} imageUpdater={this.imageUpdater} initial={this.state.initial} image={this.state.image} loadingChecker={this.loadingChecker} closePopover={this.closePopover}/>

        </CSSTransition>

      </div>
    );
  }
}

export default Main;