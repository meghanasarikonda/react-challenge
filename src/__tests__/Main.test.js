import React from 'react';
import ReactDOM from 'react-dom';
import Main from '../components/Main';
import Enzyme, { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';


function setup() {
 
  const enzymeWrapper = mount(<Main />)
 
  return {
    enzymeWrapper
  }
}

describe('Main', () => {

  it('should render correctly', () => {
    const {enzymeWrapper} = setup()
    expect(shallowToJson(enzymeWrapper)).toMatchSnapshot();
  });

  it('should have 4 key-value pairs in state', () => {
    const {enzymeWrapper} = setup()
    expect(Object.keys(enzymeWrapper.state()).length).toBe(4);
  });

  it('should have SingleAvatar as child', () => {
    const {enzymeWrapper} = setup()
    expect(enzymeWrapper.find('SingleAvatar').length).toBe(1);
  });

  it('should have AvatarList as child', () => {
    const {enzymeWrapper} = setup()
    expect(enzymeWrapper.find('AvatarList').length).toBe(0);
    //enzymeWrapper.find('Main').at(2).simulate('click');
    // console.log(enzymeWrapper.find('CSSTransition').find('AvatarList'), '<<<<<<<<')
    enzymeWrapper.setState({ openAvatarList: true });
    expect(enzymeWrapper.find('AvatarList').length).toBe(1);
  });

  it('should have image as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('image')).toBe(true);
  });

  it('should have ImageList as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('ImageList')).toBe(true);
  });

  it('should have openAvatarList as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('openAvatarList')).toBe(true);
  });

  it('should have isLoading as key in state', () => {
    const {enzymeWrapper} = setup();
    expect(enzymeWrapper.state().hasOwnProperty('isLoading')).toBe(true);
  });

  it('should have closebox function', () => {
    const {enzymeWrapper} = setup();
    enzymeWrapper.setState({ openAvatarList: true });
    expect(typeof enzymeWrapper.instance().closebox).toBe('function');
    enzymeWrapper.instance().closebox();
    expect(enzymeWrapper.state().openAvatarList).toBe(false);
  });

  it('should have loadingChecker method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().loadingChecker).toBe('function');
  });

  it('should have openPopover method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().openPopover).toBe('function');
  });

  it('should have imageUpdater method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().imageUpdater).toBe('function');
  });

  it('should have SingleAvatar method', () => {
    const {enzymeWrapper} = setup();
    expect(typeof enzymeWrapper.instance().singleavatar).toBe('function');
  })

})