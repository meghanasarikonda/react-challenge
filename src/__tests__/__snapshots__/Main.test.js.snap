// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render correctly 1`] = `
<Main>
  <div>
    <SingleAvatar
      image={
        Object {
          "id": 1,
          "label": "Avatar 1",
          "src": "avatar1.png",
        }
      }
      openPopover={[Function]}
    >
      <div>
        <img
          alt="Avatar 1"
          className="avatar"
          key="1"
          onClick={[Function]}
          src="avatar1.png"
        />
      </div>
    </SingleAvatar>
    <CSSTransition
      classNames="box"
      in={false}
      timeout={300}
      unmountOnExit={true}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={true}
      />
    </CSSTransition>
  </div>
</Main>
`;
